File: 3731-find-missing-elements/3731-find-missing-elements.ts
function findMissingElements(nums: number[]): number[] {
    const present: boolean[] = new Array(101).fill(false);
    let minVal = Infinity, maxVal = -Infinity;

    for (const ele of nums) {
        minVal = Math.min(minVal, ele);
        maxVal = Math.max(maxVal, ele);
        present[ele] = true;
    }

    const res: number[] = [];
    for (let i = minVal; i <= maxVal; i++) {
        if (!present[i]) res.push(i);
    }
    return res;
}

File: 3731-find-missing-elements/3731-find-missing-elements.rs
impl Solution {
    pub fn find_missing_elements(nums: Vec<i32>) -> Vec<i32> {
        let mut present = [false; 101];
        let mut min_val = i32::MAX;
        let mut max_val = i32::MIN;

        for &ele in &nums {
            min_val = min_val.min(ele);
            max_val = max_val.max(ele);
            present[ele as usize] = true;
        }

        let mut res = Vec::new();
        for i in min_val..=max_val {
            if !present[i as usize] {
                res.push(i);
            }
        }
        res
    }
}